Keen.NetStandard/Dataset/Datasets.cs

```diff
@@ -0,0 +1,306 @@
+﻿using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using Newtonsoft.Json.Serialization;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Threading.Tasks;
+
+
+namespace Keen.Core.Dataset
+{
+    /// <summary>
+    /// Datasets implements the IDataset interface which represents the Keen.IO Cached Datasets
+    /// API methods.
+    /// </summary>
+    internal class Datasets : IDataset
+    {
+        private const int MAX_DATASET_DEFINITION_LIST_LIMIT = 100;
+        private static readonly JsonSerializerSettings SERIALIZER_SETTINGS =
+            new JsonSerializerSettings
+            {
+                ContractResolver = new DefaultContractResolver
+                {
+                    NamingStrategy = new SnakeCaseNamingStrategy()
+                },
+                DefaultValueHandling = DefaultValueHandling.IgnoreAndPopulate,
+                Formatting = Formatting.None
+            };
+
+        private readonly IKeenHttpClient _keenHttpClient;
+        private readonly string _cachedDatasetRelativeUrl;
+        private readonly string _masterKey;
+
+
+        internal Datasets(IProjectSettings prjSettings,
+                          IKeenHttpClientProvider keenHttpClientProvider)
+        {
+            if (null == prjSettings)
+            {
+                throw new ArgumentNullException(nameof(prjSettings),
+                                                "Project Settings must be provided.");
+            }
+
+            if (null == keenHttpClientProvider)
+            {
+                throw new ArgumentNullException(nameof(keenHttpClientProvider),
+                                                "A KeenHttpClient provider must be provided.");
+            }
+
+            if (string.IsNullOrWhiteSpace(prjSettings.KeenUrl) ||
+                !Uri.IsWellFormedUriString(prjSettings.KeenUrl, UriKind.Absolute))
```

Comment:
Yeah I'm actually not sure why I didn't do that during PR #51 or whenever it was that I added these lines of code to `Queries`/et. al.

Reply:
I was playing with this and realized that part of the problem is you can implement `IProjectSettings` in 3rd party code. That said, in theory since our impls of `Datasets`/`Queries`/`Event`/etc. have `internal` ctors, you can't directly construct them, so we *could* just call it sufficient to validate the project settings in the `KeenClient` ctor. However, I think this is better off being fixed as part of Issue #56 since we should consolidate a bunch of the code in all those impls.